src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/HyperdriveOffsetComparisonService.scala

```diff
@@ -0,0 +1,196 @@
+  override def getResolvedAppArguments(jobDefinition: ResolvedJobDefinition): Option[Map[String, String]] = {
+    if (isHyperdriveJob(jobDefinition)) {
+      logger.warn(s"Job Definition ${jobDefinition} is not a Hyperdrive Job!")
+      None
+    } else {
+      val jobParameters = jobDefinition.jobParameters.asInstanceOf[SparkInstanceParameters]
+      val args = jobParameters.appArguments
+      val config = parseConfiguration(args.toArray)
+      import scala.collection.JavaConverters._
+      val resolvedArgs = config.getKeys.asScala.map { k =>
+        k -> config.getString(k)
+      }.toMap
+      Some(resolvedArgs)
+    }
+  }
+
+  override def getHdfsParameters(resolvedAppArguments: Map[String, String]): Option[HdfsParameters] = {
+    val hdfsParameters = for {
+      keytab <- sparkConfig.yarn.additionalConfs.get("spark.yarn.keytab")
+      principal <- sparkConfig.yarn.additionalConfs.get("spark.yarn.principal")
+      checkpointLocation <- resolvedAppArguments.get(HyperdriveCheckpointKey)
+    } yield new HdfsParameters(keytab, principal, checkpointLocation)
+
+    if (hdfsParameters.isEmpty) {
+      logger.warn(
+        s"Could not extract hdfs parameters from spark config ${sparkConfig}" +
+          s" and resolved app arguments ${resolvedAppArguments}"
+      )
+    }
+
+    hdfsParameters
+  }
+
+  override def getKafkaParameters(jobDefinition: ResolvedJobDefinition): Option[(String, Properties)] = {
+    if (isHyperdriveJob(jobDefinition)) {
+      logger.warn(s"Job Definition ${jobDefinition} is not a Hyperdrive Job!")
+      None
+    } else {
+      val args = jobDefinition.jobParameters.asInstanceOf[SparkInstanceParameters].appArguments
+      val kafkaParameters = for {
+        topic <- args
+          .find(_.startsWith(s"$HyperdriveKafkaTopicKey="))
+          .map(_.replace(s"$HyperdriveKafkaTopicKey=", ""))
+        brokers <- args
+          .find(_.startsWith(s"$HyperdriveKafkaBrokersKey="))
+          .map(_.replace(s"$HyperdriveKafkaBrokersKey=", ""))
+        extraArgs = args
+          .filter(_.startsWith(s"$HyperdriveKafkaExtraOptionsKey."))
+          .map(_.replace(s"$HyperdriveKafkaExtraOptionsKey.", ""))
+          .filter(_.contains("="))
+          .map { s =>
+            val keyValue = s.split("=", 2)
+            val key = keyValue(0).trim
+            val value = keyValue(1).trim
+            (key, value)
+          }
+          .toMap
+      } yield {
+        val properties = new Properties()
+        properties.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokers)
+        extraArgs.foreach { case (key, value) => properties.setProperty(key, value) }
+        (topic, properties)
+      }
+
+      if (kafkaParameters.isEmpty) {
+        logger.warn(
+          s"Could not find required kafka parameters in job definition ${jobDefinition.name} with args ${args}"
+        )
+      }
+      kafkaParameters
+    }
+  }
+
+  private def isHyperdriveJob(jobDefinition: ResolvedJobDefinition) =
```

Comment:
I don't get this method. 
Should not it be obDefinition.jobParameters.jobType == JobTypes.Hyperdrive &&
      jobDefinition.jobParameters.isInstanceOf[SparkInstanceParameters]

Reply:
It should be `isNotHyperdriveJob` 🤦

Reply:
Done